### PR TITLE
RMDX

### LINKED ISSUE #1494: Create logic to generate contentful-like AST from source MDX

### PR DESCRIPTION
**Pull request by jharvey10**
This is the full and self-contained implementation of the RMDX parser.

Closes #1494

This supersedes part of #1497

### CODE REVIEW

File: scripts/add-pr-comment.mjs [hunk 1]
```diff
@@ -193,4 +193,5 @@ const resultText = getResultText(updatedWorkspaces)
 const response = await addPrComment(resultText)
 console.log(response)
 
+console.log()
```
Comment (francinelucca) [inline scripts/add-pr-comment.mjs:196]: is this intentional?

File: packages/rmdx/src/main/process.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+type NodeHandlers = Record<keyof typeof nodeHandlers, NodeHandler>
+
+const processor = createProcessor({
+  remarkPlugins: [remarkGfm]
+})
+
+/**
+ * The actual top-level visitor/node handler.
+ *
+ * @param data kwArgs input data
```
Comment (francinelucca) [inline packages/rmdx/src/main/process.ts:39]: is `kwArgs` here intentional? 😅

**Review by francinelucca — APPROVED** [review]
this looks great! 🚀

File: packages/rmdx/README.md [hunk 3]
```diff
@@ -0,0 +1,133 @@
+# RMDX (Remote MDX)
+
+A set of utilities that allow remote MDX to be safely rendered as a component tree in a react app.
```
Comment (andreancardona) [inline packages/rmdx/README.md:3]:
```suggestion
A set of utilities that allow remote MDX to be safely rendered as a component tree in a React app.
```

**Review by andreancardona — APPROVED** [review]

File: scripts/add-pr-comment.mjs (hunk 1 above)
Comment (jdharvey-ibm) [reply scripts/add-pr-comment.mjs:196]: Yeah, just an extra newline I wanted in there. I could switch it to `console.log(response + '\n')` if you think that'd make more sense.

File: packages/rmdx/src/main/process.ts (hunk 2 above)
Comment (jdharvey-ibm) [reply packages/rmdx/src/main/process.ts:39]: Yeah, I can expand out that abbreviation though to "keyword args" or something more specific.

Comment (jdharvey-ibm) [reply packages/rmdx/src/main/process.ts:39]: Done!

**Pull request merged**